enterprise/mqtt/src/test/java/io/crate/mqtt/netty/MqttNettyNetworkConfigTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.mqtt.netty;
+
+import io.crate.action.sql.SQLOperations;
+import io.crate.ingestion.IngestionService;
+import io.crate.metadata.Functions;
+import io.crate.protocols.postgres.BindPostgresException;
+import io.crate.protocols.ssl.SslContextProvider;
+import io.crate.settings.SharedSettings;
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.user.StubUserManager;
+import org.elasticsearch.common.network.NetworkService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.http.BindHttpException;
+import org.elasticsearch.transport.BindTransportException;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.net.UnknownHostException;
+import java.util.Collections;
+
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.MQTT_ENABLED_SETTING;
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.resolvePublishPort;
+import static java.net.InetAddress.getByName;
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class MqttNettyNetworkConfigTest extends CrateUnitTest {
+
+    @Test
+    public void testPSQLPublishPort() throws Exception {
```

Comment:
should probably be called `testMqttPublishPort` instead of `PSQL` - ideally the test method should also give an indication of what aspect exactly it tests.

Reply:
+1